### PR TITLE
feat: added file-size-limit feature

### LINKED ISSUE #50: Limit file size to upload in Course banner image
### Issue Description

Right now users can upload any size of images as their banner cover image

### Steps to Reproduce

Watch this jam video: https://jam.dev/c/2044bdfc-638f-4e1c-b0dd-0f5f4a903946

image

### Expected Behavior

1. Click upload image
2. If more than 500kb don't upload to supabase and show error
3. Text button should change to try again
4. If user clicks try again and same issue, go back to step 2
5. else upload the image to supabase and the existing flow should remain the same

Also make sure to update the text saying file limit is 500KB

### Current Behavior

Any file size can be uploaded

### Screenshots (optional)

_No response_

### Operating System [e.g., Windows 10]

All

### Browser [e.g., Chrome, Firefox]

All

### Environment (locally, production, selfhosted)

All

### Please check the boxes that apply to this issue report.

- [X] I have searched the existing issues to make sure this is not a duplicate.
- [X] I have provided steps to reproduce the issue.
- [X] I have included relevant environment information.
- [X] I have included any relevant screenshots.
- [ ] I understand that this is a voluntary contribution and that there is no guarantee of resolution.
- [ ] I want to work on creating a PR for this issue if approved

### PR DESCRIPTION
**Pull request by Toyin5**
## What does this PR do?

Limit the file upload for course banner cover picture to 500kb

Fixes #50 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- I tested it on my browser(localhost)

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

**Comment by Toyin5** [comment]
@rotimi-best review pls

**Comment by rotimi-best** [comment]
@Toyin5 Can you merge main branch into yours?

File: supabase/migrations/20231115082347_remote_schema.sql [hunk 1]
```diff
@@ -1258,11 +1258,11 @@ INSERT INTO "public"."question_type" (label, created_at, updated_at, typename) V
 INSERT INTO "public"."question_type" (label, created_at, updated_at, typename) VALUES ('Paragraph', '2021-08-07 18:49:46.246529+00', '2021-08-15 00:57:38.634665+00', 'TEXTAREA');
 
 
-alter table "storage"."buckets" drop constraint "buckets_owner_fkey";
+-- alter table "storage"."buckets" drop constraint "buckets_owner_fkey";
 
-alter table "storage"."buckets" add column "owner_id" text;
+-- alter table "storage"."buckets" add column "owner_id" text;
 
-alter table "storage"."objects" add column "owner_id" text;
```
Comment (rotimi-best) [inline supabase/migrations/20231115082347_remote_schema.sql]: I didn't notice this, can you please undo this change.

**Comment by rotimi-best** [comment]
@Toyin5 Thanks a lot for it.

Looks good to me 🎉 

Merging in now
